### DISCUSSION TITLE
useFetch and POST data

### DISCUSSION THREAD

**Discussion by Babelfisch** [vueuse/vueuse#632] [Q&A]
I am a little bit confused about how `useFetch` works. With this simple call I try to submit a POST request to a PHP script:

```javascript
const { data } = useFetch(url).post({ foo: "bar" });
```

But the PHP script doesn't receive the POST data and `$_POST['foo']` is not set.

There is no example with data in the doc and I tried many variants without success. Any ideas how I can send POST data to a script?

Second question: In the doc and several discussion, the usage of `json()` or `.text()` is like this:

```javascript
const { data } = useFetch(url).get().json()
```

But this order does not work and produces this error:

```
Uncaught TypeError: Object(...)(...).get(...).json is not a function
```

Only this works by me:

```javascript
const { data } = useFetch(url).json().get()
```

Whats correct and why?

**Comment by isgj** [comment]
Hi @Babelfisch 

When you send the post request
```ts
const { data } = useFetch(url).post({ foo: "bar" });
```
what do you see on the network tab of the browser?

---

This is correct, the docs should be updated.

```ts
const { data } = useFetch(url).json().get()
```
Even when `useFetch` will make the call immediately it makes the real call in a `setTimeout(..., 0)` so we can change some defaults with ease like `useFetch('/rest').json().post({...})`. Since `get` is the default http method `useFetch` will use, in your case is enough
```ts
const { data } = useFetch(url).json()
```

**Reply by xorik** [reply]
Because it's how JSON API works. It never goes directly to $_POST, you need to get body and decode it manually. Or use tools in framwork (if you use one).

For example in Symfony you can use this bundle: https://github.com/symfony-bundles/json-request-bundle, it will convert json fields and add it to the Response object
